// osu-api-extended/dist/api/v2/routes/scores/download.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
exports.description = {
    auth: 0,
    title: __filename,
    method: 'GET',
    description: 'Download score replay file',
    params: [
        {
            type: 'number',
            name: 'score_id',
            optional: false,
            description: 'id of the score',
        },
        {
            type: 'string',
            name: 'mode',
            optional: false,
            description: '\`\`\`osu\`\`\` or \`\`\`fruits\`\`\` or \`\`\`mania\`\`\` or \`\`\`taiko\`\`\`',
        },
        {
            type: 'string',
            name: 'file_path',
            optional: false,
            description: 'File path with \`\`\`.osr\`\`\` at the end',
        },
    ],
    return: 'string',
};
const name = async (score_id, mode, file_path) => {
    const data = await (0, request_1.download)(`https://osu.ppy.sh/api/v2/scores/${mode}/${score_id}/download`, file_path);
    return data;
};
exports.default = name;
