// osu-api-extended/dist/api/v2/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = exports.chat = exports.notifications = exports.room = exports.rooms = exports.matches = exports.site = exports.comments = exports.changelogs = exports.assets = exports.forums = exports.beatmaps = exports.beatmap = exports.scores = exports.users = exports.user = void 0;
const activity_1 = __importStar(require("./routes/user/activity"));
const details_1 = __importStar(require("./routes/user/details"));
const details_2 = __importStar(require("./routes/user/me/details"));
const friends_1 = __importStar(require("./routes/user/me/friends"));
const quota_1 = __importStar(require("./routes/user/me/download/quota"));
const category_1 = __importStar(require("./routes/user/beatmaps/category"));
const most_played_1 = __importStar(require("./routes/user/beatmaps/most_played"));
const kudosu_1 = __importStar(require("./routes/user/beatmaps/kudosu"));
exports.user = {
    activity: activity_1.default,
    details: details_1.default,
    beatmaps: {
        kudosu: kudosu_1.default,
        category: category_1.default,
        most_played: most_played_1.default,
    },
    me: {
        details: details_2.default,
        friends: friends_1.default,
        download: {
            quota: quota_1.default
        }
    },
};
const details_3 = __importStar(require("./routes/users/details"));
const events_1 = __importStar(require("./routes/users/events"));
const list_1 = __importStar(require("./routes/users/groups/list"));
exports.users = {
    details: details_3.default,
    events: events_1.default,
    groups: {
        list: list_1.default,
    },
};
const details_4 = __importStar(require("./routes/scores/details"));
const download_1 = __importStar(require("./routes/scores/download"));
const category_2 = __importStar(require("./routes/scores/user/category"));
const beatmap_1 = __importStar(require("./routes/scores/user/beatmap"));
const beatmap_2 = __importStar(require("./routes/scores/beatmap"));
exports.scores = {
    details: details_4.default,
    download: download_1.default,
    beatmap: beatmap_2.default,
    user: {
        category: category_2.default,
        beatmap: beatmap_1.default,
    },
};
const details_5 = __importStar(require("./routes/beatmap/id/details"));
const attributes_1 = __importStar(require("./routes/beatmap/id/attributes"));
const lookup_1 = __importStar(require("./routes/beatmap/id/lookup"));
const details_6 = __importStar(require("./routes/beatmap/set/details"));
const lookup_2 = __importStar(require("./routes/beatmap/set/lookup"));
const download_2 = __importStar(require("./routes/beatmap/set/download"));
const addToFavourites_1 = __importStar(require("./routes/beatmap/set/addToFavourites"));
const details_7 = __importStar(require("./routes/beatmap/discussions/details"));
const posts_1 = __importStar(require("./routes/beatmap/discussions/posts"));
const votes_1 = __importStar(require("./routes/beatmap/discussions/votes"));
exports.beatmap = {
    id: {
        attributes: attributes_1.default,
        lookup: lookup_1.default,
        details: details_5.default,
    },
    set: {
        lookup: lookup_2.default,
        details: details_6.default,
        download: download_2.default,
        addToFavourites: addToFavourites_1.default,
    },
    discussions: {
        votes: votes_1.default,
        posts: posts_1.default,
        details: details_7.default,
    },
};
const details_8 = __importStar(require("./routes/beatmaps/details"));
const search_1 = __importStar(require("./routes/beatmaps/search"));
const events_2 = __importStar(require("./routes/beatmaps/events"));
exports.beatmaps = {
    details: details_8.default,
    search: search_1.default,
    events: events_2.default,
};
const details_9 = __importStar(require("./routes/forums/topic/details"));
const edit_1 = __importStar(require("./routes/forums/topic/edit"));
const edit_2 = __importStar(require("./routes/forums/post/edit"));
const new_1 = __importStar(require("./routes/forums/topic/new"));
const reply_1 = __importStar(require("./routes/forums/topic/reply"));
// import forums_topic_vote, { description as forums_topic_vote_desc } from "./routes/forums/topic/vote";
//
exports.forums = {
    topic: {
        new: new_1.default,
        edit: edit_1.default,
        reply: reply_1.default,
        // vote: forums_topic_vote,
        details: details_9.default,
    },
    post: {
        edit: edit_2.default
    },
};
const seasonalBackgrounds_1 = __importStar(require("./routes/assets/seasonalBackgrounds"));
exports.assets = {
    seasonalBackgrounds: seasonalBackgrounds_1.default,
};
const list_2 = __importStar(require("./routes/changelogs/list"));
const lookup_3 = __importStar(require("./routes/changelogs/lookup"));
const details_10 = __importStar(require("./routes/changelogs/details"));
exports.changelogs = {
    list: list_2.default,
    lookup: lookup_3.default,
    details: details_10.default,
};
const list_3 = __importStar(require("./routes/comments/list"));
const details_11 = __importStar(require("./routes/comments/details"));
const new_2 = __importStar(require("./routes/comments/new"));
const edit_3 = __importStar(require("./routes/comments/edit"));
const remove_1 = __importStar(require("./routes/comments/remove"));
const vote_1 = __importStar(require("./routes/comments/vote"));
exports.comments = {
    new: new_2.default,
    edit: edit_3.default,
    list: list_3.default,
    vote: vote_1.default,
    remove: remove_1.default,
    details: details_11.default,
};
const search_2 = __importStar(require("./routes/site/search"));
const wiki_1 = __importStar(require("./routes/site/wiki"));
const list_4 = __importStar(require("./routes/site/spotlights/list"));
const details_12 = __importStar(require("./routes/site/ranking/details"));
const list_5 = __importStar(require("./routes/site/news/list"));
const details_13 = __importStar(require("./routes/site/news/details"));
exports.site = {
    search: search_2.default,
    wiki: wiki_1.default,
    spotlights: {
        list: list_4.default,
    },
    ranking: {
        details: details_12.default,
    },
    news: {
        list: list_5.default,
        details: details_13.default,
    },
};
const list_6 = __importStar(require("./routes/matches/list"));
const details_14 = __importStar(require("./routes/matches/details"));
exports.matches = {
    list: list_6.default,
    details: details_14.default,
};
const list_7 = __importStar(require("./routes/rooms/list"));
exports.rooms = {
    list: list_7.default,
};
const details_15 = __importStar(require("./routes/room/details"));
const leaderboard_1 = __importStar(require("./routes/room/leaderboard"));
exports.room = {
    details: details_15.default,
    leaderboard: leaderboard_1.default,
};
const list_8 = __importStar(require("./routes/notifications/list"));
const markAsReaded_1 = require("./routes/notifications/markAsReaded");
exports.notifications = {
    list: list_8.default,
    // markAsReaded: notifications_markAsReaded, // TODO:
};
const list_9 = __importStar(require("./routes/chat/channels/list"));
const details_16 = __importStar(require("./routes/chat/channels/details"));
const join_1 = __importStar(require("./routes/chat/channels/join"));
const leave_1 = __importStar(require("./routes/chat/channels/leave"));
const new_3 = __importStar(require("./routes/chat/new"));
const updates_1 = __importStar(require("./routes/chat/updates"));
const list_10 = __importStar(require("./routes/chat/channels/messages/list"));
const send_1 = __importStar(require("./routes/chat/channels/messages/send"));
const markAsReaded_2 = __importStar(require("./routes/chat/channels/messages/markAsReaded"));
exports.chat = {
    new: new_3.default,
    updates: updates_1.default,
    channels: {
        list: list_9.default,
        join: join_1.default,
        leave: leave_1.default,
        details: details_16.default,
        messages: {
            list: list_10.default,
            send: send_1.default,
            markAsReaded: markAsReaded_2.default,
        }
    },
};
exports.description = {
    user: {
        activity: activity_1.description,
        details: details_1.description,
        beatmaps: {
            kudosu: kudosu_1.description,
            category: category_1.description,
            most_played: most_played_1.description,
        },
        me: {
            details: details_2.description,
            friends: friends_1.description,
            download: {
                quota: quota_1.description,
            },
        },
    },
    users: {
        details: details_3.description,
        events: events_1.description,
        groups: {
            list: list_1.description,
        },
    },
    scores: {
        details: details_4.description,
        download: download_1.description,
        beatmap: beatmap_2.description,
        user: {
            category: category_2.description,
            beatmap: beatmap_1.description,
        },
    },
    beatmap: {
        id: {
            attributes: attributes_1.description,
            lookup: lookup_1.description,
            details: details_5.description,
        },
        set: {
            lookup: lookup_2.description,
            details: details_6.description,
            download: download_2.description,
            addToFavourites: addToFavourites_1.description,
        },
        discussions: {
            votes: votes_1.description,
            posts: posts_1.description,
            details: details_7.description,
        },
    },
    beatmaps: {
        details: details_8.description,
        search: search_1.description,
        events: events_2.description,
    },
    forums: {
        topic: {
            new: new_1.description,
            edit: edit_1.description,
            reply: reply_1.description,
            // vote: forums_topic_vote_desc,
            details: details_9.description,
        },
        post: {
            edit: edit_2.description,
            // details: forums_topic_details_desc,
        },
    },
    assets: {
        seasonalBackgrounds: seasonalBackgrounds_1.description,
    },
    changelogs: {
        list: list_2.description,
        lookup: lookup_3.description,
        details: details_10.description,
    },
    comments: {
        new: new_2.description,
        edit: edit_3.description,
        list: list_3.description,
        vote: vote_1.description,
        remove: remove_1.description,
        details: details_11.description,
    },
    site: {
        search: search_2.description,
        wiki: wiki_1.description,
        spotlights: {
            list: list_4.description,
        },
        ranking: {
            details: details_12.description,
        },
        news: {
            list: list_5.description,
            details: details_13.description,
        },
    },
    matches: {
        list: list_6.description,
        details: details_14.description,
    },
    rooms: {
        list: list_7.description,
    },
    room: {
        details: details_15.description,
        leaderboard: leaderboard_1.description,
    },
    notifications: {
        list: list_8.description,
        markAsReaded: markAsReaded_1.description,
    },
    chat: {
        new: new_3.description,
        updates: updates_1.description,
        channels: {
            list: list_9.description,
            join: join_1.description,
            leave: leave_1.description,
            details: details_16.description,
            messages: {
                list: list_10.description,
                send: send_1.description,
                markAsReaded: markAsReaded_2.description,
            }
        },
    },
};
/*
POST /chat/ack
POST /chat/channels
GET /chat/presence

POST /rooms
GET /rooms/{room}/playlist/{playlist}/scores/users/{user}
GET /rooms/{room}/playlist/{playlist}/scores
GET /rooms/{room}/playlist/{playlist}/scores/{score}
PUT /rooms/{room}/users/{user}
DELETE /rooms/{room}/users/{user}
POST /rooms/{room}/playlist/{playlist}/scores
PUT /rooms/{room}/playlist/{playlist}/scores/{score}

POST /beatmaps/{beatmap}/solo/scores
PUT /beatmaps/{beatmap}/solo/scores/{token}

POST /reports










*/
