// osu-api-extended/dist/api/v2/routes/beatmap/attributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'POST',
    description: 'Return attributes (stars, combo, stats) of beatmap',
    params: [
        {
            type: 'number',
            name: 'beatmap_id',
            optional: false,
            description: 'Beatmap id',
        },
        {
            name: 'body',
            params: [
                {
                    type: 'string array',
                    name: 'mods',
                    optional: true,
                    description: 'Array of matching mods [\'HD\', \'DT\']',
                },
                {
                    type: 'string',
                    name: 'ruleset',
                    optional: true,
                    description: `\`\`\`osu\`\`\` or \`\`\`fruits\`\`\` or \`\`\`mania\`\`\` or \`\`\`taiko\`\`\``,
                },
                {
                    type: 'number',
                    name: 'ruleset_id',
                    optional: true,
                    description: `\`\`\`0\`\`\` or \`\`\`1\`\`\` or \`\`\`2\`\`\` or \`\`\`3\`\`\``,
                },
            ]
        },
    ],
};
;
const name = async (beatmap_id, body) => {
    const data = await request(`beatmaps/${beatmap_id}/attributes`, {
        method: 'POST',
        data: JSON.stringify(body),
    });
    return data;
};
exports.default = name;
