// osu-api-extended/dist/api/v2/routes/beatmap/download.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
exports.description = {
    auth: 0,
    title: __filename,
    method: 'GET',
    description: 'Download beatmap set to specified directory',
    params: [
        {
            type: 'number',
            name: 'beatmapset',
            optional: false,
            description: 'id of the beatmap set',
        },
        {
            type: 'string',
            name: 'file_path',
            optional: false,
            description: 'path to file with their name and extension',
        },
        {
            type: 'function',
            name: 'callback',
            optional: false,
            description: 'function which is will be triggered on downloading progress',
        },
    ],
};
;
const name = async (beatmapset, file_path, callback) => {
    const data = await (0, request_1.download)(`https://osu.ppy.sh/api/v2/beatmapsets/${beatmapset}/download`, file_path, {}, callback);
    return data;
};
exports.default = name;
