// osu-api-extended/dist/api/v2/routes/users/groups/list.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../utility/request");
// const request: RequestNamespace = namespace('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 3,
    title: __filename,
    method: 'GET',
    description: 'Return array of users from specified group id',
    params: [
        {
            type: 'number',
            name: 'id',
            optional: false,
            description: '\`\`\`4\`\`\` or \`\`\`7\`\`\` or \`\`\`11\`\`\` or \`\`\`16\`\`\` or \`\`\`22\`\`\` or \`\`\`28\`\`\` or \`\`\`31\`\`\` or \`\`\`32\`\`\`',
        },
    ],
};
const name = async (id) => {
    try {
        const data = await (0, request_1.request)(`https://osu.ppy.sh/groups/${id}`, {
            method: 'GET',
        });
        const parse = data.split('<script id="json-users" type="application/json">')[1].split('</script>')[0];
        return JSON.parse(parse);
    }
    catch (err) {
        console.log('groups-list error', id, err.message);
        return [];
    }
};
exports.default = name;
