// osu-api-extended/dist/api/v2/routes/user/beatmaps/category.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return beatmaps list of specified user',
    params: [
        {
            type: 'number',
            name: 'user',
            optional: false,
            description: 'id of the user',
        },
        {
            type: 'string',
            name: 'type',
            optional: false,
            description: '\`\`\`favourite\`\`\` or \`\`\`loved\`\`\` or \`\`\`ranked\`\`\` or \`\`\`pending\`\`\` or \`\`\`graveyard\`\`\`',
        },
        {
            name: 'object',
            params: [
                {
                    type: 'number',
                    name: 'limit',
                    optional: true,
                    description: 'Maximum number of results',
                },
                {
                    type: 'number',
                    name: 'offset',
                    optional: true,
                    description: 'Result offset for pagination',
                },
            ]
        },
    ],
};
const name = async (user, type, obj) => {
    const data = await request(`users/${user}/beatmapsets/${type}`, {
        method: 'GET',
        params: obj,
    });
    return data;
};
exports.default = name;
