// osu-api-extended/dist/api/v1/routes/user/scores-category.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const best_1 = __importDefault(require("../../form/user/best"));
const recent_1 = __importDefault(require("../../form/user/recent"));
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/');
const _mode = [
    'osu',
    'taiko',
    'fruits',
    'mania',
];
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return scores from user for a specified type',
    params: [
        {
            type: 'string',
            name: 'user',
            optional: false,
            description: 'id of the user',
        },
        {
            type: 'string',
            name: 'mode',
            optional: true,
            description: '\`\`\`osu\`\`\` or \`\`\`fruits\`\`\` or \`\`\`mania\`\`\` or \`\`\`taiko\`\`\`',
        },
        {
            type: 'string',
            name: 'type',
            optional: true,
            description: '\`\`\`u\`\`\` is a user_id or a username. Use string for usernames or \`\`\`id\`\`\` for user_ids',
        },
        {
            type: 'string',
            name: 'event_days',
            optional: true,
            description: 'Max number of days between now and last event date. Range of 1-31. Optional, default value is 1',
        },
    ],
};
;
const name = async (user, type, obj = {}) => {
    const params = {
        u: user,
        m: _mode.indexOf(obj.mode),
        type: obj.type,
        limit: obj.limit,
    };
    const data = await request(`get_user_${type}`, {
        method: 'GET',
        params: params,
    });
    if (data.length == 0)
        return null;
    if (type == 'best') {
        const format = (0, best_1.default)(data, obj.mode);
        return format;
    }
    ;
    const format = (0, recent_1.default)(data, obj.mode);
    return format;
};
exports.default = name;
