// osu-api-extended/dist/api/v2/routes/beatmap/set/addToFavourites.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 0,
    title: __filename,
    method: 'POST',
    description: 'Added specified beatmapset to favourite list',
    params: [
        {
            type: 'number',
            name: 'beatmapset_id',
            optional: false,
            description: 'id of the beatmap set',
        },
        {
            type: 'boolean',
            name: 'action',
            optional: false,
            description: 'true/false',
        },
    ],
};
const name = async (beatmapset_id, action) => {
    let _action = '';
    if (action == true)
        _action = 'favourite';
    if (action == false)
        _action = 'unfavourite';
    const data = await request(`beatmapsets/${beatmapset_id}/favourites`, {
        method: 'POST',
        params: { action: _action }
    });
    return data;
};
exports.default = name;
