// osu-api-extended/dist/api/v2/routes/_user/scores/beatmap/all.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return list of all user scores on a beatmap',
    params: [
        {
            type: 'number',
            name: 'beatmap',
            optional: false,
            description: 'id of the beatmap',
        },
        {
            type: 'string/number',
            name: 'user',
            optional: true,
            description: 'id of the user',
        },
        {
            type: 'string',
            name: 'mode',
            optional: true,
            description: '\`\`\`osu\`\`\` or \`\`\`fruits\`\`\` or \`\`\`mania\`\`\` or \`\`\`taiko\`\`\`',
        },
    ],
};
;
const name = async (beatmap, user, mode) => {
    const data = await request(`beatmaps/${beatmap}/scores/users/${user}/all`, {
        method: 'GET',
        params: { mode },
    });
    return data;
};
exports.default = name;
