// osu-api-extended/dist/api/v2/routes/_user/scores/beatmap/best.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return best user score on beatmap',
    params: [
        {
            type: 'number',
            name: 'beatmap',
            optional: false,
            description: 'id of the beatmap',
        },
        {
            type: 'string/number',
            name: 'user',
            optional: true,
            description: 'id of the user',
        },
        {
            name: 'object',
            params: [
                {
                    type: 'string',
                    name: 'mode',
                    optional: true,
                    description: '\`\`\`osu\`\`\` or \`\`\`fruits\`\`\` or \`\`\`mania\`\`\` or \`\`\`taiko\`\`\`',
                },
                {
                    type: 'string array',
                    name: 'mods',
                    optional: true,
                    description: 'Array of matching mods [\'HD\', \'DT\']',
                },
            ]
        },
    ],
};
;
const name = async (beatmap, user, obj) => {
    const data = await request(`beatmaps/${beatmap}/scores/users/${user}`, {
        method: 'GET',
        params: obj,
    });
    return data;
};
exports.default = name;
