// osu-api-extended/dist/index.js
"use strict";
// export * as v1 from "./api/v1/index";
// export * as v2 from "./api/v2/index";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.download = exports.v2 = exports.v1 = exports.auth = exports.tools = exports.mods = void 0;
exports.mods = __importStar(require("./utility/mods"));
exports.tools = __importStar(require("./utility/tools/index"));
exports.auth = __importStar(require("./utility/auth"));
exports.v1 = __importStar(require("./api/v1/index"));
exports.v2 = __importStar(require("./api/v2/index"));
var request_1 = require("./utility/request");
Object.defineProperty(exports, "download", { enumerable: true, get: function () { return request_1.download; } });
// export * as md from './md';
