// osu-api-extended/dist/api/v1/form/user/best.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const index_1 = require("../../../../utility/tools/index");
const mods_1 = require("../../../../utility/mods");
const name = (data, mode) => {
    const info = [];
    for (let i = 0; i < data.length; i++) {
        const d = data[i];
        const score = {
            date: d.date,
            beatmap: +d.beatmap_id,
            rank: d.rank,
            user: {
                id: +d.user_id,
            },
            score: {
                id: +d.score_id,
                total: +d.score,
            },
            combo: {
                max: +d.maxcombo,
                full: +d.perfect,
            },
            hits: {
                300: +d.count300,
                geki: +d.countgeki,
                100: +d.count100,
                katu: +d.countkatu,
                50: +d.count50,
                0: +d.countmiss,
            },
            mods: {
                id: +d.enabled_mods,
                name: (0, mods_1.name)(+d.enabled_mods),
            },
            accuracy: 0,
            pp: parseFloat(d.pp),
            replay: +d.replay_available,
        };
        score.accuracy = (0, index_1.accuracy)(score.hits, mode);
        info.push(score);
    }
    ;
    return info;
};
exports.default = name;
