// osu-api-extended/dist/api/v2/routes/rooms/list.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: '',
    params: [
        {
            type: 'string',
            name: 'mode',
            optional: false,
            description: '\`\`\`all\`\`\` or \`\`\`owned\`\`\` or \`\`\`participated\`\`\` or \`\`\`ended\`\`\`',
        },
        {
            name: 'object',
            optional: true,
            params: [
                {
                    type: 'string',
                    name: 'category',
                    optional: true,
                    description: '\`\`\`normal\`\`\` or \`\`\`spotlight\`\`\` or \`\`\`featured_artist\`\`\`',
                },
                {
                    type: 'number',
                    name: 'limit',
                    optional: true,
                    description: 'Number of results',
                },
                {
                    type: 'number',
                    name: 'season_id',
                    optional: true,
                    description: 'season_id',
                },
                {
                    type: 'string',
                    name: 'type_group',
                    optional: true,
                    description: '\`\`\`playlists\`\`\` or \`\`\`realtime\`\`\`',
                },
            ],
        }
    ],
};
const name = async (mode, object) => {
    //@ts-ignore
    object.mode = mode;
    const data = await request(`rooms/${mode == 'all' ? 'ended' : mode}`, {
        method: 'GET',
        params: object,
    });
    return data;
};
exports.default = name;
