// osu-api-extended/dist/api/v2/routes/changelogs/details.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Returns details of the specified build',
    params: [
        {
            type: 'string',
            name: 'stream',
            optional: false,
            description: '\`\`\`stable40\`\`\` or \`\`\`stable\`\`\` or \`\`\`beta40\`\`\` or \`\`\`cuttingedge\`\`\` or \`\`\`lazer\`\`\` or \`\`\`web\`\`\`',
        },
        {
            type: 'string',
            name: 'build',
            options: false,
            optional: false,
            description: '\`\`\`id\`\`\` or \`\`\`name\`\`\` of the build',
        },
    ],
};
const name = async (stream, build) => {
    const data = await request(`changelog/${stream}/${build}`, {
        method: 'GET',
    });
    return data;
};
exports.default = name;
