// osu-api-extended/dist/api/v2/routes/changelogs/lookup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Returns details of the specified build',
    params: [
        {
            type: 'string/number',
            name: 'changelog',
            optional: false,
            description: 'Build version, update stream name, or build ID',
        },
        {
            name: 'object',
            params: [
                {
                    type: 'string',
                    name: 'key',
                    options: false,
                    optional: false,
                    description: 'Unset to query by build version or stream name, or \`\`\`id\`\`\` to query by build ID.',
                },
                {
                    type: 'string[]',
                    name: 'message_formats',
                    optional: false,
                    description: '\`\`\`html\`\`\` or \`\`\`markdown\`\`\`',
                },
            ]
        },
    ],
};
const name = async (changelog, obj) => {
    const data = await request(`changelog/${changelog}`, {
        method: 'GET',
        params: obj,
    });
    return data;
};
exports.default = name;
