// osu-api-extended/dist/utility/tools/routes/pp/calculate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.desc = exports.description = void 0;
const request_1 = require("../../../request");
exports.description = {
    auth: 4,
    title: __filename,
    method: 'GET',
    description: 'Return pp and details about beatmap',
    params: [
        {
            type: 'string',
            name: 'id',
            optional: false,
            description: 'id of the beatmap',
        },
        {
            type: 'string/number',
            name: 'mods',
            optional: true,
            description: 'Name of the mods \`\`\`HDDT\`\`\` or mods number \`\`\`72\`\`\`',
        },
        {
            type: 'string/number',
            name: 'combo',
            optional: true,
            description: 'Max combo of the play',
        },
        {
            type: 'number',
            name: 'miss',
            optional: true,
            description: 'Amount of misses',
        },
        {
            type: 'number',
            name: 'acc',
            optional: true,
            description: 'Amount of accuracy',
        },
    ],
};
;
const name = async (id, mods, combo, miss, acc) => {
    const data = await (0, request_1.request)('https://pp.osuck.net/pp', { method: 'GET', params: { id, mods, combo, miss, acc } });
    if (!data.id)
        return null;
    return data;
};
exports.default = name;
exports.desc = exports.description;
