// osu-api-extended/dist/utility/tools/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.desc = exports.pp = exports.download = exports.total_objects = exports.accuracy = exports.country = exports.rank = void 0;
const rank_1 = __importStar(require("./routes/rank"));
const country_1 = __importStar(require("./routes/country"));
const accuracy_1 = __importStar(require("./routes/accuracy"));
const total_objects_1 = __importStar(require("./routes/total_objects"));
const difficulty_1 = __importStar(require("./routes/download/difficulty"));
const calculate_1 = __importStar(require("./routes/pp/calculate"));
exports.rank = rank_1.default;
exports.country = country_1.default;
exports.accuracy = accuracy_1.default;
exports.total_objects = total_objects_1.default;
exports.download = {
    difficulty: difficulty_1.default
};
exports.pp = {
    calculate: calculate_1.default
};
exports.desc = {
    rank: rank_1.description,
    country: country_1.description,
    accuracy: accuracy_1.description,
    total_objects: total_objects_1.description,
    download: {
        difficulty: difficulty_1.description,
    },
    pp: {
        calculate: calculate_1.description,
    },
};
