// osu-api-extended/dist/api/v2/routes/beatmap/discussions/posts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return posts from beatmap set discussions',
    params: [
        {
            name: 'object',
            params: [
                {
                    type: 'number',
                    name: 'beatmapset_discussion_id',
                    optional: true,
                    description: 'id of beatmap set id',
                },
                {
                    type: 'number',
                    name: 'limit',
                    optional: true,
                    description: 'Maximum number of results',
                },
                {
                    type: 'number',
                    name: 'page',
                    optional: true,
                    description: 'Search page',
                },
                {
                    type: 'string',
                    name: 'sort',
                    optional: true,
                    description: '\`\`\`id_desc\`\`\` or \`\`\`id_asc\`\`\`',
                },
                {
                    type: 'string[]',
                    name: 'types',
                    optional: true,
                    description: '\`\`\`first\`\`\` or \`\`\`replay\`\`\` or \`\`\`system\`\`\`',
                },
                {
                    type: 'number',
                    name: 'user',
                    optional: true,
                    description: 'id of the user',
                },
            ],
        },
    ],
};
const name = async (obj) => {
    const data = await request(`beatmapsets/discussions/posts`, {
        method: 'GET',
        params: obj,
    });
    return data;
};
exports.default = name;
