// osu-api-extended/dist/api/v2/routes/site/news/list.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Returns a list of news posts and related metadata',
    params: [
        {
            name: 'object',
            params: [
                {
                    type: 'number',
                    name: 'limit',
                    optional: true,
                    description: 'Maximum number of posts (12 default, 1 minimum, 21 maximum)',
                },
                {
                    type: 'number',
                    name: 'year',
                    optional: true,
                    description: 'Year to return posts from',
                },
                {
                    type: 'string',
                    name: 'cursorPublished',
                    optional: true,
                    description: 'Pagination cursorPublished',
                },
                {
                    type: 'number',
                    name: 'cursorId',
                    optional: true,
                    description: 'Pagination cursorId',
                },
            ],
        }
    ],
    return: 'response',
};
const name = async (obj) => {
    const data = await request(`news`, {
        method: 'GET',
        params: obj,
    });
    return data;
};
exports.default = name;
