// osu-api-extended/dist/api/v2/routes/site/search.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Searches users and wiki pages.',
    params: [
        {
            name: 'object',
            params: [
                {
                    type: 'string',
                    name: 'mode',
                    optional: true,
                    description: '\`\`\`all\`\`\` or \`\`\`user\`\`\` or \`\`\`wiki_page\`\`\`',
                },
                {
                    type: 'string',
                    name: 'query',
                    optional: true,
                    description: 'Search keyword',
                },
                {
                    type: 'number',
                    name: 'page',
                    optional: true,
                    description: 'Search result page. Ignored for mode all',
                },
            ],
        }
    ],
    return: 'response',
};
const name = async (obj) => {
    const data = await request(`search`, {
        method: 'GET',
        params: obj,
    });
    return data;
};
exports.default = name;
