// osu-api-extended/dist/api/v2/routes/beatmap/id/lookup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return data of the specified beatmap id',
    params: [
        {
            name: 'object',
            params: [
                {
                    type: 'number',
                    name: 'id',
                    optional: true,
                    description: 'id of the beatmap',
                },
                {
                    type: 'string',
                    name: 'checksum',
                    optional: true,
                    description: 'md5 of the beatmap file',
                },
                {
                    type: 'string',
                    name: 'filename',
                    optional: true,
                    description: 'file name of the beatmap',
                },
            ],
        },
    ],
};
const name = async (object) => {
    const data = await request(`beatmaps/lookup`, {
        method: 'GET',
        params: {
            checksum: object.checksum,
            filename: object.filename,
            id: object.id,
        },
    });
    return data;
};
exports.default = name;
