// osu-api-extended/dist/api/v1/routes/match.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../utility/request");
const match_1 = __importDefault(require("../form/match"));
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return match details',
    params: [
        {
            type: 'string',
            name: 'id',
            optional: false,
            description: 'id of the user',
        },
    ],
};
;
const name = async (id) => {
    const params = { mp: id };
    const data = await request(`get_match`, {
        method: 'GET',
        params: params,
    });
    if (data.match == 0 || !data.match)
        return null;
    const format = (0, match_1.default)(data);
    return format;
};
exports.default = name;
