// osu-api-extended/dist/api/v2/routes/user/activity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const request_1 = require("../../../../utility/request");
const request = (0, request_1.namespace)('https://osu.ppy.sh/api/v2/');
exports.description = {
    auth: 1,
    title: __filename,
    method: 'GET',
    description: 'Return list of recent user activity',
    params: [
        {
            type: 'number',
            name: 'user',
            optional: false,
            description: 'id of the user',
        },
        {
            name: 'object',
            params: [
                {
                    type: 'number',
                    name: 'limit',
                    optional: true,
                    description: 'Maximum number of results',
                },
                {
                    type: 'string',
                    name: 'offset',
                    optional: true,
                    description: 'Result offset for pagination',
                },
            ]
        },
    ],
};
const name = async (user, object = {}) => {
    const data = await request(`users/${user}/recent_activity`, {
        method: 'GET',
        params: object,
    });
    return data;
};
exports.default = name;
